### PR TITLE
Update altCLIP model card

### PR DESCRIPTION
**Pull request by EmileAydar**
# What does this PR do?

As suggested in this issue - https://github.com/huggingface/transformers/issues/36979#issuecomment-2902300855 - this PR rewrites the altCLIP model card so it matches the standardized template that @stevhliu introduced in #36979.

Key points:

- Friend-style overview + sidebar badges.
- <hfoptions> usage tabs — Pipeline tab explains why it’s missing, AutoModel tab contains a runnable similarity example, transformers-cli tab shows one-liner quantisation.
- Quantisation section with dynamic INT-8 snippet (Linear-only, Embeddings left FP32) and link to the Quantization guide.
- Attention visualisation section: explains current AttentionMaskVisualizer limitation and shows a manual ViT-attention workaround; embeds a PNG heat-map (CLS attention, last layer, head 
- All text follows Steve’s template verbatim where required.

Check list

- [DONE] Brief model description
- [ DONE] Ready-to-use code examples (Pipeline ✗, AutoModel ✓, transformers-cli ✓)
- [ DONE] Quantisation example for a large model (dynamic INT-8)
- [DONE ] Attention mask visualiser (note + workaround ) # I did not provide an image because it is not produced from the HF visualizer, but I can if required.

# Who can review?
Hi @stevhliu !

I updated the altCLIP model card following the standard format. 

Let me know what you think of it.

Happy to revise. Thanks!

### CODE REVIEW

File: docs/source/en/model_doc/altclip.md [hunk 1]
```diff
@@ -1,116 +1,131 @@
...
-*In this work, we present a conceptually simple and effective method to train a strong bilingual multimodal representation model. 
-Starting from the pretrained multimodal representation model CLIP released by OpenAI, we switched its text encoder with a pretrained 
-multilingual text encoder XLM-R, and aligned both languages and image representations by a two-stage training schema consisting of 
-teacher learning and contrastive learning. We validate our method through evaluations of a wide range of tasks. We set new state-of-the-art 
-performances on a bunch of tasks including ImageNet-CN, Flicker30k- CN, and COCO-CN. Further, we obtain very close performances with 
-CLIP on almost all tasks, suggesting that one can simply alter the text encoder in CLIP for extended capabilities such as multilingual understanding.*
+<hfoption id="AutoModel">
 
-This model was contributed by [jongjyh](https://huggingface.co/jongjyh).
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md:36]: Update this snippet to:

```py
import torch
import requests
from PIL import Image
from transformers import AltCLIPModel, AltCLIPProcessor

model = AltCLIPModel.from_pretrained("BAAI/AltCLIP", torch_dtype=torch.bfloat16)
processor = AltCLIPProcessor.from_pretrained("BAAI/AltCLIP")

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities

labels = ["a photo of a cat", "a photo of a dog"]
for label, prob in zip(labels, probs[0]):
    print(f"{label}: {prob.item():.4f}")
```

File: docs/source/en/model_doc/altclip.md [hunk 2]
```diff
@@ -1,116 +1,131 @@
...
+<hfoption id="Pipeline">
 
-The AltCLIP model was proposed in [AltCLIP: Altering the Language Encoder in CLIP for Extended Language Capabilities](https://arxiv.org/abs/2211.06679v2) by Zhongzhi Chen, Guang Liu, Bo-Wen Zhang, Fulong Ye, Qinghong Yang, Ledell Wu. AltCLIP
-(Altering the Language Encoder in CLIP) is a neural network trained on a variety of image-text and text-text pairs. By switching CLIP's
-text encoder with a pretrained multilingual text encoder XLM-R, we could obtain very close performances with CLIP on almost all tasks, and extended original CLIP's capabilities such as multilingual understanding.
+`pipeline()` isn’t available because altCLIP currently exposes only its contrastive head (image × text similarity) and no dedicated *visual-question-answering* or *captioning* head.  
+Use the AutoModel path shown below instead.
 
-The abstract from the paper is the following:
+</hfoption>
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md:32]: You can remove the `Pipeline` section here

File: docs/source/en/model_doc/altclip.md [hunk 3]
```diff
@@ -1,116 +1,131 @@
...
+# one-liner via transformers-cli
+transformers-cli quantize BAAI/AltCLIP \
+  --method dynamic \
+  --dtype int8 \
+  --modules Linear \
+  --output AltCLIP-int8
 ```
 
-<Tip>
+Or do it in plain PyTorch:
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md:84]: Replace with just one simple example:

```py
# !pip install torchao
import torch
import requests
from PIL import Image
from transformers import AltCLIPModel, AltCLIPProcessor, TorchAoConfig

model = AltCLIPModel.from_pretrained("BAAI/AltCLIP", quantization_config = TorchAoConfig("int4_weight_only", group_size=128, torch_dtype=torch.bfloat16)
processor = AltCLIPProcessor.from_pretrained("BAAI/AltCLIP")

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)

inputs = processor(text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True)

outputs = model(**inputs)
logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities

labels = ["a photo of a cat", "a photo of a dog"]
for label, prob in zip(labels, probs[0]):
    print(f"{label}: {prob.item():.4f}")
```

File: docs/source/en/model_doc/altclip.md [hunk 4]
```diff
@@ -1,116 +1,131 @@
...
-## AltCLIPVisionConfig
+> ℹ️ Embedding layers can be quantized with *float-qparams weight-only* configs once PyTorch exposes them via the public API.
 
-[[autodoc]] AltCLIPVisionConfig
 
-## AltCLIPProcessor
+---
 
-[[autodoc]] AltCLIPProcessor
+## Attention visualisation
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md:111]: This can be removed since it isn't supported

File: docs/source/en/model_doc/altclip.md [hunk 5]
```diff
@@ -1,116 +1,131 @@
...
+- altCLIP’s ViT backbone uses **16 × 16 pixel patches**, so a 224 × 224 image becomes a 14 × 14 patch grid.  
+  When you reshape attention scores remember to use that grid size.
 
-[[autodoc]] AltCLIPVisionModel
-    - forward
\ No newline at end of file
+```py
+grid_len = int((attn[-1].size(-1) - 1) ** 0.5)  # 14 for ViT-B/16
+heatmap  = attn[-1][0, 0, 1:, 0].reshape(grid_len, grid_len)
+```
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md:131]: Add a `## Notes` section:

```suggestion
## Notes

- AltCLIP uses bidirectional attention instead of causal attention and it uses the `[CLS]` token in XLM-R to represent a text embedding.
- Use [`CLIPImageProcessor` to resize (or rescale) and normalize images for the model.
- [`AltCLIPProcessor`] combines [`CLIPImageProcessor`] and [`XLMRobertaTokenizer`] into a single instance to encode text and prepare images.
```

**Review by stevhliu — COMMENTED** [review]
Thanks, added some suggestions to make it simpler!

File: docs/source/en/model_doc/altclip.md [hunk 6]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
 
-# AltCLIP
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+<div style="float: right;">
+  <div class="flex flex-wrap space-x-1">
+    <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+    <img alt="Transformers" src="https://img.shields.io/badge/Transformers-6B5B95?style=flat&logo=transformers&logoColor=white">
+  </div>
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

File: docs/source/en/model_doc/altclip.md [hunk 7]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
 
-This model was contributed by [jongjyh](https://huggingface.co/jongjyh).
+The examples below demonstrates how to calculate similarity scores between an image and one or more captions with the [`AutoModel`] class.
 
-## Usage tips and example
+<hfoptions id="usage">
 
-The usage of AltCLIP is very similar to the CLIP. the difference between CLIP is the text encoder. Note that we use bidirectional attention instead of casual attention
-and we take the [CLS] token in XLM-R to represent text embedding.
+<hfoption id="Pipeline">
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]: You can remove the entire `Pipeline` section

File: docs/source/en/model_doc/altclip.md [hunk 8]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
+
+<hfoption id="transformers-cli">
+
+AltCLIP does **not** require `transformers-cli` at inference time, but the tool is handy for quantisation (see next section).
+
+</hfoption>
+
+</hfoptions>
+
+---
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

File: docs/source/en/model_doc/altclip.md [hunk 9]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
 
-This model is based on `CLIPModel`, use it like you would use the original [CLIP](clip).
+## Quantization
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

File: docs/source/en/model_doc/altclip.md [hunk 10]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
+logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
+probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
 
-## AltCLIPVisionConfig
+labels = ["a photo of a cat", "a photo of a dog"]
+for label, prob in zip(labels, probs[0]):
+    print(f"{label}: {prob.item():.4f}")
+```
+
+On a typical machine the INT-8 checkpoint occupies **≈ ½ the RAM** of the full-precision model with negligible accuracy drop.
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

File: docs/source/en/model_doc/altclip.md [hunk 11]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
 
-[[autodoc]] AltCLIPVisionConfig
+> ℹ️ Embedding layers can be quantized with *float-qparams weight-only* configs once PyTorch exposes them via the public API.
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

File: docs/source/en/model_doc/altclip.md [hunk 12]
```diff
@@ -14,103 +14,126 @@ rendered properly in your Markdown viewer.
...
 
-## AltCLIPProcessor
 
-[[autodoc]] AltCLIPProcessor
+---
```
Comment (stevhliu) [inline docs/source/en/model_doc/altclip.md]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Thanks, there are still some unresolved comments. You also don't need to modify the other files `testing_utils.py` and `modeling_deit.py`.

**Comment by EmileAydar** [comment]
Hi @stevhliu, 
The checks seem to be all valid now. 
Please let me know if you’d like any further tweaks or if you think this is good to go.

**Review by stevhliu — APPROVED** [review]
Thanks!
